From: Robert Schwebel <r.schwebel@pengutronix.de>
Date: Sat, 12 Feb 2011 22:21:18 +0100
Subject: [PATCH] fix issues with newer compilers

The fam packet has several issues with newer gcc compilers, which are
fixed by this patch.

Signed-off-by: Robert Schwebel <r.schwebel@pengutronix.de>
---
 include/BTree.h |    2 ++
 lib/Client.c++  |    2 +-
 src/IMon.c++    |    2 +-
 src/Interest.h  |    2 ++
 4 files changed, 6 insertions(+), 2 deletions(-)

diff --git a/include/BTree.h b/include/BTree.h
index 29ea94f..f978bda 100644
--- a/include/BTree.h
+++ b/include/BTree.h
@@ -23,6 +23,8 @@
 #ifndef BTree_included
 #define BTree_included
 
+#include <cstdlib>
+
 #include "Boolean.h"
 
 //  This is an in-core B-Tree implementation.
diff --git a/lib/Client.c++ b/lib/Client.c++
index fa62eb4..c3c4631 100644
--- a/lib/Client.c++
+++ b/lib/Client.c++
@@ -34,7 +34,7 @@
 #include <syslog.h>
 #include <errno.h>
 
-#include <iostream.h>
+#include <iostream>
 
 #include "fam.h"
 #include "Client.h"
diff --git a/src/IMon.c++ b/src/IMon.c++
index ea53a49..2dcbed0 100644
--- a/src/IMon.c++
+++ b/src/IMon.c++
@@ -40,7 +40,7 @@
 #include "Interest.h"
 #include "Log.h"
 #include "Scheduler.h"
-#include "alloc.h"
+#include <cstdlib>
 
 int		   IMon::imonfd = -2;
 IMon::EventHandler IMon::ehandler = NULL;
diff --git a/src/Interest.h b/src/Interest.h
index af95256..8ae551d 100644
--- a/src/Interest.h
+++ b/src/Interest.h
@@ -23,6 +23,8 @@
 #ifndef Interest_included
 #define Interest_included
 
+#include <cstdlib>
+
 #include <sys/stat.h>
 #include <sys/types.h>
 #include <sys/time.h>
-- 
1.7.2.3

